### PR TITLE
[Issue-27424] Include UTM parameters in announcement urls

### LINKED ISSUE #27424: Include UTM Parameters in Announcement URLs
### Parent Issue

#25733 

### Task

We should enhance the application by including UTM parameters in the URLs for announcements. The UTM parameters will provide valuable information for tracking the source, medium, and campaign associated with each announcement. The desired format for the UTM parameters is ?utm_source=platform&utm_medium={HOST_NAME}&utm_campaign=announcement. This enhancement will improve analytics and tracking for announcements on the platform.

### Proposed Objective

Same as Parent Issue

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

_No response_

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by manuelrojas**
### Proposed Changes

Enhancing the announcements menu functionality by incorporating UTM parameters into the associated links. Additionally, ensuring dynamic updating of these links whenever there is a change in the site, providing a seamless and consistent user experience.

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/dot-toolbar-announcements.component.html [hunk 1]
```diff
@@ -23,19 +23,21 @@ <h5 class="announcements__title">{{ 'announcements' | dm }}</h5>
...
     }}</a>
 </div>
 
 <h5 class="announcements__title">{{ 'announcements_knowledge_center' | dm }}</h5>
 <div class="announcements__about">
-    <a *ngFor="let link of knowledgeCenterLinks" [href]="link.url" target="_blank">{{
-        link.label
-    }}</a>
+    @for (item of knowledgeCenterLinks(); track item.id;) {
+    <a [href]="item.url" target="_blank">{{ item.label }}</a>
```
Comment (zJaaal) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/dot-toolbar-announcements.component.html:34]: SonarQube probably will throw an error here. You should add `rel="noopener"` to make it pass the SQ test

File: core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/dot-toolbar-announcements.component.html [hunk 2]
```diff
@@ -23,19 +23,21 @@ <h5 class="announcements__title">{{ 'announcements' | dm }}</h5>
...
+    }
 </div>
 
 <h5 class="announcements__title">{{ 'announcements_knowledge_contact_us' | dm }}</h5>
 <div class="announcements__about">
-    <a *ngFor="let link of contactLinks" [href]="link.url" target="_blank">{{ link.label }}</a>
+    @for (item of contactLinks(); track item.id;) {
+    <a [href]="item.url" target="_blank">{{ item.label }}</a>
```
Comment (zJaaal) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/dot-toolbar-announcements.component.html:41]: same

File: core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/store/dot-announcements.store.ts [hunk 3]
```diff
@@ -70,6 +80,45 @@ export class AnnouncementsStore extends ComponentStore<DotAnnouncementsState> {
...
+                label: this.dotMessageService.get('announcements.contact.customer.support'),
+                url: `https://www.dotcms.com/services/support/?${state.utmParameters}`
+            },
+            {
+                id: '2',
+                label: this.dotMessageService.get('announcements.contact.professional.services'),
+                url: `https://www.dotcms.com/services/professional-services/?${state.utmParameters}`
+            }
+        ];
+    });
```
Comment (zJaaal) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/store/dot-announcements.store.ts:116]: SQ can see this as duplicated code, you can change the order of the properties in the object to make it pass the tests

**Review by zJaaal — APPROVED** [review]
Just watch out of SQ, you can resolve my comments

**Review by rjvelazco — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/dot-toolbar-announcements.component.ts [hunk 4]
```diff
@@ -17,41 +22,32 @@ import { DotMessagePipe } from '@dotcms/ui';
...
-        {
-            label: 'Customer Support',
-            url: 'https://www.dotcms.com/services/support/?utm_source=dotcms&utm_medium=application&utm_campaign=announcement_menu'
-        },
-        {
-            label: 'Professional Services',
-            url: 'https://www.dotcms.com/services/professional-services/?utm_source=dotcms&utm_medium=application&utm_campaign=announcement_menu'
-        }
-    ];
+        this.siteService.switchSite$.pipe(skip(1)).subscribe(() => {
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-toolbar/components/dot-toolbar-announcements/dot-toolbar-announcements.component.ts:40]: Unsubscribe?

**Review by hmoreras — APPROVED** [review]

**Pull request merged**